airbyte-integrations/connectors/source-uk-police/source_uk_police/spec.yaml

```diff
@@ -0,0 +1,56 @@
+documentationUrl: https://docsurl.com
+connectionSpecification:
+  $schema: http://json-schema.org/draft-07/schema#
+  title: UK Police Spec
+  type: object
+  required:
+    - start_month
+    - lat
+    - lng
+  additionalProperties: true
+  properties:
+    start_month:
+      type: string
+      description: The month to start from
+      examples:
+        - 2019-01
+      pattern: "^[0-9]{4}-[0-9]{2}$"
+      order: 0
+    end_month:
+      type: string
+      description: The month to end at
+      examples:
+        - 2019-01
+      pattern: "^[0-9]{4}-[0-9]{2}$"
+      order: 1
+    crime_category:
+      type: string
+      description: |
+        The crime category to filter by. See the /crime-categories endpoint
+        for a list of valid categories. Defaults to 'all-crime'.
+      examples:
+        - anti-social-behaviour
+        - burglary
+      order: 2
+    lat:
+      type: number
+      description: Latitude of the centre of the search area.
+      examples:
+        - 51.509865
+      order: 3
+    lng:
+      type: number
+      description: Longitude of the centre of the search area.
+      examples:
+        - -0.118092
+      order: 4
+    force:
+      type: string
+      description: |
+        The force to filter by. See the /forces endpoint for a list of valid
+        forces. Required for the '*_no_location' and '*_by_force' streams.
+      examples:
+        - avon-and-somerset
+        - city-of-london
+      order: 5
```

Comment:
Why you don't make those streams substream of force one and retrieve all data to all forces?

Reply:
Similar to the MonthStreamSlicer you can create one for Forces.

Reply:
I could, but would that not mean I wouldn't be able to use the `DatetimeStreamSlicer` for those streams? I don't believe I can use both

Reply:
@girarda is it possible to interpolate two StreamSlicer?

Reply:
@isaacharrisholt you can wrap in a [cartesian product stream slicers](https://docs.airbyte.com/connector-development/config-based/understanding-the-yaml-file/stream-slicers#cartesian-product-stream-slicer) to iterate over all the pairs (force, datetime)

Reply:
@girarda I've made that change, but I'm getting the following when I run the check operation:

```
{"type": "LOG", "log": {"level": "ERROR", "message": "Check failed"}}
{"type": "CONNECTION_STATUS", "connectionStatus": {"status": "FAILED", "message": "\"Unable to connect to stream crimes_with_no_location - 'NoneType' object has no attribute 'get'\""}}
```

I've pushed the changes to the repo in commit [2664eb8](https://github.com/airbytehq/airbyte/pull/18344/commits/2664eb8903b139e109db2efd4e3852e2bbe0401b)

Reply:
@girarda can you check this change?

Reply:
found the [bug here](https://github.com/airbytehq/airbyte/pull/18529/files#diff-9328dc3f2f6ae4ce085ddc1e7da60a9c02002e64d9311ca907f07abf411409d4R89)

@isaacharrisholt to unblock yourself, you can add do the same None check and use your local version of the CDK.

you can do this by running the following commands from the connector's directory:
```
pip uninstall airbyte_cdk
pip install -e ../../../airbyte-cdk/python/
```

I tried reading from the stream, but I'm getting a 404, but I also get reach the sample request from their doc (https://data.police.uk/api/crimes-no-location?category=all-crime&force=leicestershire&date=2017-03)

Let me know if you have more questions!

Reply:
@girarda I'm also getting a 404, but it seems like the `force` request parameter isn't being properly injected.

```
{"type": "DEBUG", "message": "Making outbound API request", "data": {"request_body": "None", "headers": "{'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'}", "url": "https://data.police.uk/api/crimes-no-location?lat=51.5074&lng=-0.1278&category=burglary&date=2022-08"}}
```

This could be an issue with my configuration, but I don't believe so?